test/Microsoft.AspNetCore.Mvc.Core.Test/ModelBinding/ModelBinderFactoryTest.cs

```diff
@@ -235,6 +236,109 @@ public void CreateBinder_Caches_WhenTokenIsNotNull()
+                    PropertyFilterProvider = propertyFilterProvider,
+                };
+
+                return new TheoryData<BindingInfo, BindingMetadata, BindingInfo>
+                {
+                    { emptyBindingInfo, emptyBindingMetadata, emptyBindingInfo },
+                    { fullBindingInfo, emptyBindingMetadata, fullBindingInfo },
+                    { emptyBindingInfo, fullBindingMetadata, fullBindingInfo },
+                    // Resulting BindingInfo combines two inputs
+                    { halfBindingInfo, secondHalfBindingMetadata, fullBindingInfo },
+                    // Parameter information has precedence over type metadata
+                    { fullBindingInfo, differentBindingMetadata, fullBindingInfo },
+                };
+            }
+        }
+
+        [Theory]
+        [MemberData(nameof(BindingInfoData))]
+        public void CreateBinder_PassesExpectedBindingInfo(
+            BindingInfo parameterBindingInfo,
+            BindingMetadata bindingMetadata,
+            BindingInfo expectedInfo)
+        {
+            // Arrange
+            var metadataProvider = new TestModelMetadataProvider();
+            metadataProvider.ForType<Employee>().BindingDetails(binding =>
+            {
+                binding.BinderModelName = bindingMetadata.BinderModelName;
+                binding.BinderType = bindingMetadata.BinderType;
+                binding.BindingSource = bindingMetadata.BindingSource;
+                if (bindingMetadata.PropertyFilterProvider != null)
+                {
+                    binding.PropertyFilterProvider = bindingMetadata.PropertyFilterProvider;
+                }
+            });
+
+            var binder = Mock.Of<IModelBinder>();
+            var options = new TestOptionsManager<MvcOptions>();
+            options.Value.ModelBinderProviders.Add(new TestModelBinderProvider(context =>
+            {
+                Assert.Equal(typeof(Employee), context.Metadata.ModelType);
+
+                Assert.NotNull(context.BindingInfo);
+                Assert.Equal(expectedInfo.BinderModelName, context.BindingInfo.BinderModelName, StringComparer.Ordinal);
+                Assert.Equal(expectedInfo.BinderType, context.BindingInfo.BinderType);
+                Assert.Equal(expectedInfo.BindingSource, context.BindingInfo.BindingSource);
+                Assert.Same(expectedInfo.PropertyFilterProvider, context.BindingInfo.PropertyFilterProvider);
+
+                return binder;
+            }));
+
+            var factory = new ModelBinderFactory(metadataProvider, options);
+            var factoryContext = new ModelBinderFactoryContext
+            {
+                BindingInfo = parameterBindingInfo,
+                Metadata = metadataProvider.GetMetadataForType(typeof(Employee)),
+            };
+
+            // Act
+            var result = factory.CreateBinder(factoryContext);
+
+            // Assert
+            Assert.Same(binder, result);
```

Comment:
This misleads the purpose of the test. I'd just leave out the assert clause here and group it above `// Act & Assert`. In the end you're just asserting your mock returned the expected value and that it flowed as the result of `CreateBinder` (not what this test is trying to test)